src/domain/Tables/Table/subcomponents/TableRow.tsx

```diff
@@ -28,80 +24,58 @@ import {
   StyledSwipeActionsButtonWrapper,
   StyledSwipeActionsCell
 } from '../services/style'
-import { IColumnProps, IRowProps, ISelectedRows, Table, getActionsIconButtonGroup } from '../Table'
+import {
+  IColumnProps,
+  IRowProps,
+  InteractionType
+} from '../services/types'
+import { getActionsIconButtonGroup } from './ActionIconButtonGroup'
 import { TableCheckboxInput, TableCheckboxInputValue } from './TableCheckboxInput'
+import { TableRowDataCell } from './TableRowDataCell'
 
 interface ITableRowProps <T> {
   columns: Array<IColumnProps<T>>
   row: IRowProps<T>
-  selectedAll: TableCheckboxInputValue
-  selectedRows: ISelectedRows
-  setSelectedRows: (value: ISelectedRows) => void
-  interactionType: Table.InteractionType
+  isSelected: boolean
+  toggleSelected: (row: IRowProps<T>) => void
+  interactionType: InteractionType
   hasLeftAction: boolean
   hasTableSwipeActions: boolean
   onRowRemove?: (removedRowData: T) => void
   expandedSwipeCellRow: string | null
   setExpandedSwipeCellRow: (value: string | null) => void
   lastRow: boolean
-  transitionState?: 'entering' | 'entered' | 'exiting' | 'exited'
-}
-
-interface ITableRowDataCellProps <T> {
-  column: IColumnProps<T>
-  row: IRowProps<T>
-  hasProgressBar: boolean
-  isLastColumn: boolean
-  hasTableSwipeActions: boolean
-  hasSwipeActions: boolean
-  selectedRows: ISelectedRows
-  setSelectedRows: (value: ISelectedRows) => void
-  setHasHovered: (value: boolean) => void
-  handleSwipeAction?: () => void
-  isFirstColumn: boolean
-  children: JSX.Element
 }
 
 interface IUseDragProps {
   /* The movement in pixel [x-axis, y-axis]*/
-  _movement: number[]
+  _movement: [number, number]
   /* If yes, the drag action is ongoing*/
   dragging: boolean
 }
 
-const TableCellTooltip: React.FC<{tooltipText: string}> = ({ tooltipText, children}) => {
-  const toggleComponent = ({ openMenu, closeMenu, toggleComponentRef, ariaProps }: any) => (
-    <span
-      onMouseEnter={openMenu}
-      onMouseLeave={closeMenu}
-      ref={toggleComponentRef}
-      {...ariaProps}
-    >
-      {children}
-    </span>
-  )
-
-  return (
-    <TooltipPopover
-      variant={TooltipPopoverVariant.Dark}
-      toggleComponent={toggleComponent}
-    >
-      {tooltipText}
-    </TooltipPopover>
-  )
+interface IGetLeftCell <T extends {}> {
+  isSelectable: boolean
+  isRemovable: boolean
+  hasLeftAction: boolean
+  row: IRowProps<T>
+  isSelected: boolean
+  toggleSelected: (row: IRowProps<T>) => void
+  onRowRemove?: (data: T) => void
 }
 
-const getLeftCell = <T extends {}>(isSelectable: boolean, isRemovable: boolean, hasLeftAction: boolean, row: IRowProps<T>, selectedRows: ISelectedRows, setSelectedRows: (value: ISelectedRows) => void, onRowRemove?: (data: T) => void ) => {
+const getLeftCell = <T extends {}>({ isSelectable, isSelected, isRemovable, hasLeftAction, row, onRowRemove, toggleSelected }: IGetLeftCell<T>) => {
```

Comment:
Moved these to be objects rather than positional arguments (positional arguments are a v bad idea after about 2 of them)